### PR TITLE
Add coptr/index tool

### PR DESCRIPTION
**Pull request by ramirobarrantes**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by ramirobarrantes** [comment]
@nf-core-bot fix linting

File: modules/nf-core/coptr/index/main.nf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    output:
+    tuple val(meta), path("*.genomes" ), emit: genome
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/main.nf:21]: Align equal signs

File: modules/nf-core/coptr/index/main.nf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        fastafolder \
+	$prefix
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        coptr: \$(coptr |& sed -E '11!d ; s/CoPTR.*?\\(v(.*?)\\).*/\\1/')
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/main.nf:38]: As above

File: modules/nf-core/coptr/index/meta.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      identifier: biotools:coptr
+
+input:
+  - - meta:
+        type: map
+        description: |
+          Groovy Map containing genome information
+    - indexfasta:
+        type: file
+        description: Fasta file(s) ending in [.fasta, .fna, .fa]
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/meta.yml:26]: Can you make it slightly more descriptive?

File: modules/nf-core/coptr/index/meta.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        ontologies:
+          - edam: "http://edamontology.org/format_1929"
+
+output:
+  - genome:
+      - meta:
+          type: map
+          description: |
+            Groovy Map containing genome information
+          ontologies: &id001
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/meta.yml:37]: The whole ontologies entry is wrong (and not needed for meta)

File: modules/nf-core/coptr/index/meta.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            - edan: "http://edamontology.org/data_3210"
+      - '*.genomes" ':
+          type: map
+          description: |
+            Groovy Map containing genome information
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/meta.yml:42]: This description is not for this file

File: modules/nf-core/coptr/index/meta.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+          ontologies: *id001
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/meta.yml:43]: The above ontologies entry should go here, the id string is unnecessary

**Review by itrujnara — CHANGES_REQUESTED** [review]
Picky comments for module code, the meta file requires some improvements

**Comment by ramirobarrantes** [comment]
@nf-core-bot fix linting

**Comment by ramirobarrantes** [comment]
@nf-core-bot fix linting

**Comment by ramirobarrantes** [comment]
@nf-core-bot fix linting

File: modules/nf-core/coptr/index/meta.yml [hunk 7]
```diff
@@ -19,8 +19,7 @@ tools:
 input:
   - - meta:
         type: map
-        description: |
-          Groovy Map containing genome information
+        description: Name of the reference genome index
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/meta.yml:22]: Change this back, I meant the indexfasta description. Meta description is standardized.

File: modules/nf-core/coptr/index/meta.yml (hunk 3 above)
Comment (itrujnara) [reply modules/nf-core/coptr/index/meta.yml:25]: Same thing again

File: modules/nf-core/coptr/index/meta.yml [hunk 8]
```diff
@@ -29,24 +28,22 @@ input:
...
-  - genome:
+  - index:
       - meta:
           type: map
-          description: |
-            Groovy Map containing genome information
-          ontologies: &id001
-            - edan: "http://edamontology.org/data_3210"
-      - '*.genomes" ':
+          description: name of the reference genome index
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/meta.yml:34]: As above, don't change the meta description

File: modules/nf-core/coptr/index/meta.yml [hunk 9]
```diff
@@ -29,24 +28,22 @@ input:
...
+      - "bowtie2":
           type: map
-          description: |
-            Groovy Map containing genome information
-          ontologies: *id001
+          description: index genome
+          ontologies:
+            - edan: "http://edamontology.org/data_3210"
```
Comment (itrujnara) [inline modules/nf-core/coptr/index/meta.yml:39]:
```suggestion
            - edam: "http://edamontology.org/data_3210"
```

File: modules/nf-core/coptr/map/main.nf [hunk 10]
```diff
@@ -0,0 +1,59 @@
+process COPTR_MAP {
```
Comment (itrujnara) [inline modules/nf-core/coptr/map/main.nf:1]: Can you please move all coptr/map code to a different PR and ping me there? Will be easier for me to review

**Review by itrujnara — CHANGES_REQUESTED** [review]
Thanks for addressing some of the comments. There are still some issues in the meta.yml. I have also suggested splitting the two modules into separate PRs for convenience.

**Comment by ramirobarrantes** [comment]
@nf-core-bot fix linting

**Review by itrujnara — APPROVED** [review]
Nice job! Good to go on this one. Please let me know when you create the other PR.

**Pull request merged**